src/data/value/ast_value.rs

```diff
@@ -73,7 +73,57 @@ impl TryFrom<&AstValue> for Value {
                 .map_or_else(|_| v.parse::<f64>().map(Value::F64), |v| Ok(Value::I64(v)))
                 .map_err(|_| ValueError::FailedToParseNumber.into()),
             AstValue::Boolean(v) => Ok(Value::Bool(*v)),
+            AstValue::SingleQuotedString(v) => Ok(Value::Str(v.to_string())),
             _ => Err(ValueError::SqlTypeNotSupported.into()),
         }
     }
 }
+
+pub fn is_same_as_data_type_ast_value(value: &&AstValue, data_type: &DataType) -> bool {
+    let is_same = matches!(
+        (data_type, value),
+        (DataType::Boolean, AstValue::Boolean(_))
+            | (DataType::Text, AstValue::SingleQuotedString(_))
+            | (DataType::Float(_), AstValue::Number(_))
+    );
+    if !is_same {
+        if let (DataType::Int, AstValue::Number(value)) = (data_type, value) {
+            matches!(value.find('.'), None) // YUCK!
+        } else {
+            false
+        }
+    } else {
+        true
+    }
+}
+
+pub fn cast_ast_value(value: AstValue, data_type: &DataType) -> Result<AstValue> {
+    match (data_type, value) {
+        (DataType::Boolean, AstValue::SingleQuotedString(value))
+        | (DataType::Boolean, AstValue::Number(value)) => Ok(match value.to_uppercase().as_str() {
+            "TRUE" | "1" => Ok(AstValue::Boolean(true)),
+            "FALSE" | "0" => Ok(AstValue::Boolean(false)),
+            _ => Err(ValueError::ImpossibleCast),
+        }?),
+        (DataType::Int, AstValue::Number(value)) => Ok(AstValue::Number(
+            value
+                .parse::<f64>()
+                .map_err(|_| ValueError::UnreachableImpossibleCast)?
+                .trunc()
+                .to_string(),
+        )),
+        (DataType::Int, AstValue::SingleQuotedString(value))
+        | (DataType::Float(_), AstValue::SingleQuotedString(value)) => Ok(AstValue::Number(value)),
+        (DataType::Int, AstValue::Boolean(value))
+        | (DataType::Float(_), AstValue::Boolean(value)) => Ok(AstValue::Number(
+            (if value { "1" } else { "0" }).to_string(),
+        )),
+        (DataType::Float(_), AstValue::Number(value)) => Ok(AstValue::Number(value)),
+        (DataType::Text, AstValue::Boolean(value)) => Ok(AstValue::SingleQuotedString(
+            (if value { "TRUE" } else { "FALSE" }).to_string(),
+        )),
+        (DataType::Text, AstValue::Number(value)) => Ok(AstValue::SingleQuotedString(value)),
+        (_, AstValue::Null) => Ok(AstValue::Null),
+        _ => Err(ValueError::UnimplementedCast.into()),
+    }
+}
```

Comment:
Let's just move this function to `evaluated.rs`.
Hmm.. it would be better to find a different place soon.
However, currently all pure util functions which handles `AstValue` only are in `evaluated.rs`.

Reply:
Done :-)
Moved to /data/ast_value.rs